Code/Framework/AzToolsFramework/AzToolsFramework/Entity/PrefabEditorEntityOwnershipService.cpp

```diff
@@ -624,13 +625,21 @@ namespace AzToolsFramework
                         }
                     }
                     AZ::ScriptSystemRequestBus::Broadcast(&AZ::ScriptSystemRequests::GarbageCollect);
+
+                    // This is a workaround until the replacement for GameEntityContext is done
+                    AzFramework::GameEntityContextEventBus::Broadcast(&AzFramework::GameEntityContextEventBus::Events::OnGameEntitiesReset);
                 });
             m_playInEditorData.m_entities.Clear();
-
-            // This is a workaround until the replacement for GameEntityContext is done
-            AzFramework::GameEntityContextEventBus::Broadcast(&AzFramework::GameEntityContextEventBus::Events::OnGameEntitiesReset);
         }
 
+        // Game entity cleanup is queued onto the next tick via the DespawnEntities call.
+        // To avoid both game entities and Editor entities active at the same time
+        // we flush the tick queue to ensure the game entities are cleared first.
+        // The Alert callback that follows the DespawnEntities call will then reactivate the editor entities
+        // This should be considered temporary as a move to a less rigid event sequence that supports async entity clean up
+        // is the desired direction forward.
+        AZ::TickBus::ExecuteQueuedEvents();
```

Comment:
Suggestion:  The blast radius could potentially be narrowed a bit further by adding a callable "ProcessQueue" or something to the SpawnableEntitiesManager and calling that directly.  By calling the TickBus, presumably some number of other functions are also getting called that might not be needed or desired.

Reply:
(Discussed offline)
In general I agree with @mbalfour-amzn in that it's better to be specific when calling functions instead of doing a wide blast. In this case though I have reservations about making the `ProcessQueue` function public as it opens the door for spamming that call during gameplay. It would be much better imho if there was an AZ::Interface in AzToolsFramework to expose that `ProcessQueue` call so it's limited to just tools where calling it makes sense. Perhaps this can be submitted as-is and @sconel can explore options for the AzToolsFramework interface approach.